core/py/src/braintrust_core/json.py

```diff
@@ -0,0 +1,60 @@
+import dataclasses
+import json
+from typing import Any, Dict, Optional
+
+from pydantic import BaseModel
```

Comment:
I don't think we can assume the user's environment has `pydantic` (nor do we depend on it). We should probably make this a dynamic import and gracefully fail.